lib/doc/worksheet.js

```diff
@@ -782,8 +783,32 @@ class Worksheet {
     this._media = value.media.map(medium => new Image(this, medium));
     this.sheetProtection = value.sheetProtection;
     this.tables = value.tables.reduce((tables, table) => {
-      const t = new Table();
-      t.model = table;
+      /**
+       * Once we have the table information and the sheet we can add the rows related to it.
+       * We parse the start and end via the reference and then we iterate over all the
+       * cells the sheet has to see if they are inside the table to add their values to its rows.
+       */
+      const [,startColumnStr, startRowStr, endColumnStr, endRowStr] = /(\D+)(\d+):(\D+)(\d+)/i.exec(table.tableRef);
+      const startColumn = startColumnStr.toUpperCase().charCodeAt(0) - 64;
+      const endColumn = endColumnStr.toUpperCase().charCodeAt(0) - 64;
+      const startRow = parseInt(startRowStr, 10);
+      const endRow = parseInt(endRowStr, 10);
+      const offset = table.headerRow ? 1 : 0;
+      const rows = [];
+
+      for(let i = (startRow + offset); i <= endRow; i++){
+        const rowValues = [];
+        that.getRow(i).eachCell((cell, colNumber) =>{
+          if(colNumber >= startColumn && colNumber <= endColumn){
+            rowValues.push(cell.value);
+          }
+        });
+      }
+
+      table.rows = rows;
+
+      // Need to pass `that` so the table has a sheet associated
+      const t = new Table(that, table);
```

Comment:
I don't see here changes for Table, that earlier hasn't any arguments: [LINE 785](https://github.com/exceljs/exceljs/pull/1222/files#diff-1724d4057ec4f981a0e6910631878ad7L785)

Reply:
I'm adding the rows to `table` in [line 808](https://github.com/exceljs/exceljs/pull/1222/files#diff-1724d4057ec4f981a0e6910631878ad7R808).

From what I understood of the code (and what I vaguely remember, it's been a while since I wrote this), I needed to pass `that` to the constructor to the new table to have the right internal references to the worksheet and such (I'm inside a `reduce` so `this` is not what I'm expecting and otherwise we didn't have access to the values) and also the new table model with the row information (i.e. `table`).